primitives/src/communities.rs

```diff
@@ -229,20 +229,36 @@ impl Location {
 	}
 }
 
-#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
+#[derive(
+	Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, PartialOrd, Ord, TypeInfo, MaxEncodedLen,
+)]
 #[cfg_attr(feature = "serde_derive", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "serde_derive", serde(rename_all = "camelCase"))]
-pub struct UnboundedCommunityMetadata {
-	/// utf8 encoded name
-	pub name: UnboundedPalletString,
-	/// utf8 encoded abbreviation of the name
-	pub symbol: UnboundedPalletString,
-	/// IPFS cid to assets necessary for community branding
-	pub assets: IpfsCid,
-	/// ipfs cid for style resources
-	pub theme: Option<IpfsCid>,
-	/// optional link to a community site
-	pub url: Option<UnboundedPalletString>,
+pub enum AnnouncementSigner {
+	Bip340([u8; 32]),
+}
```

Comment:
If we'd use the Account Id here, we'd get base58 encoding in the common UIs, this will be reflected as a hex. I am uncertain if this is what we want, but I wanted to mention this.

Reply:
this will be a nostr pubkey and needs no substrate UI encoding

Reply:
and accountid doesn't support bip340

Reply:
I agree with the first, so absolutely fine with me, but yeah I was rather thinking of the `AccountId32`, which is agnostic to the sig-scheme not the  `T::AccountId`, which has to satisfy some trait bounds. However, I agree that we can leave it like that. In the future, we might want to have a:

```
struct Bip340/NostrPub([u8; 32]); 
```

to implement some methods on it, but this is a non-breaking change in terms of encoding and can also wait.